lib/utils/compose.coffee

```diff
@@ -201,7 +203,8 @@ exports.buildProject = (
 		renderer = new BuildProgressUI(tty, imageDescriptors)
 	renderer.start()
 
-	qemu.installQemuIfNeeded(emulated, logger, arch)
+	Promise.resolve(checkBuildSecretsRequirements(docker, projectPath))
+	.then qemu.installQemuIfNeeded(emulated, logger, arch)
```

Comment:
Does this work? `.then` will expect a function, and `installQemuIfNeeded` returns a promise. It may work but please make sure. Likely you just need to add a `->` after `.then`.

Reply:
Good catch! It was not intentional, it was missing `->`. I had tested `balena build` and `balena deploy` which ran this code and the build / deploy were successful, but presumably I had not tested Qemu emulation as `balena-secrets` is not compatible with a standard docker daemon...

Reply:
I was also not sure if this was a bluebird promise, and I could totally see bluebird accepting something like that.